### ISSUE TITLE
windows 0.12 beta3 crash on discard

### ISSUE THREAD

**Issue by tomhughes** [tomhughes/trac-tickets#1195]
**[Submitted to the original trac issue database at 2.53am, Tuesday, 23rd September 2008]**

Twice I've done some editing, gotten confused where I don't like the changes made and then I try to exit Merkartor.  

Then Merkaartor prompts to discard the data, then crashes on clicking Discard.

Both times I've looked around and tried undo first and when that didn't help tried discard.

Because I don't want the changes, this doesn't lose data; but programs shouldn't segment fault.

Attached is a drwatson dump with the last two crashes on discard.

Further back in the log are more crashes of beta2.  This was related to heavy scrolling with Yahoo on the screen; I haven't seen this yahoo scrolling crash in beta3

**Comment by tomhughes** [comment]
**Author: koying**
**[Added to the original trac issue at 10.59am, Wednesday, 24th September 2008]**

Those crashes at closing time have always the same cause.
Sometime during the session, a feature was deleted but there is still a pointer to it.
When Merkaartor is doing the cleanup before closing, it crashes with a protection exception.

Unfortunately, the root cause, i.e. where was the feature deleted, is almost impossible to find out without a complete history of the session.

Bottom line: The only ways to correct this is to have an MDC just before the crash or for myself to experiment it while under debugging...

**Comment by tomhughes** [comment]
**Author: koying**
**[Added to the original trac issue at 11.31am, Wednesday, 3rd December 2008]**

This should be solved in 0.13 by the passage to QObject - guarded pointers for features
